### PR TITLE
added packages/conf-g++/conf-g++.1.0

### PR DESCRIPTION
**Pull request by UnixJunkie**

### CODE REVIEW

**Comment by UnixJunkie** [comment]
Looks like it is working.
Someone might want to edit the PR and fill in the nixos line, if they know what it should be.

**Comment by mseri** [comment]
Thanks. I'll leave it open a few days in case somebody knows what to put there

**Comment by kit-ty-kate** [comment]
On nixos it seems to be `gcc` (https://nixos.org/nixos/packages.html#gcc)

However I'm not a fan of enforcing `g++` as the default `c++` compiler. Why not just call the package `conf-g++` ?

**Comment by ygrek** [comment]
this package looks confusing indeed. E.g. on debian there is already alternatives system, and there is `c++` symlink available that should be used instead of hardcoding g++ (unless g++ specifically is required for build of course)

**Comment by UnixJunkie** [comment]
I fixed all suggested things.

File: packages/conf-g++/conf-g++.1.0/opam [hunk 1]
```diff
@@ -0,0 +1,22 @@
+opam-version: "2.0"
+maintainer: "unixjunkie@sdf.org"
+authors: "Francois Berenger"
+homepage: "https://github.com/ocaml/opam-repository"
+bug-reports: "https://github.com/ocaml/opam-repository/issues"
+dev-repo: "git+https://github.com/ocaml/opam-repository.git"
```
Comment (dra27) [inline packages/conf-g++/conf-g++.1.0/opam]: There's no need for `dev-repo` for conf packages.

Comment (UnixJunkie) [reply packages/conf-g++/conf-g++.1.0/opam]: but opam lint gives me a warning if I remove it

**Comment by kit-ty-kate** [comment]
Thanks a lot!

File: packages/conf-g++/conf-g++.1.0/opam (hunk 1 above)
Comment (dra27) [reply packages/conf-g++/conf-g++.1.0/opam]: Really? What version of opam are you running. `opam lint` does not give a warning for no `dev-repo` if there's no `url`.

Comment (dra27) [reply packages/conf-g++/conf-g++.1.0/opam]: I’m not an opam-repository maintainer so, no, I couldn’t edit your PR...

Comment (dra27) [reply packages/conf-g++/conf-g++.1.0/opam]: Incidentally, the change to `opam lint` was made for the release candidate 8 months ago - you should upgrade!

**Pull request merged**